packages/legend-query/src/stores/QueryBuilderPropertyEditorState.ts

```diff
@@ -185,17 +268,19 @@ const fillDerivedPropertyArguments = (
       ),
       ...propertyArguments,
     ],
-    queryBuilderState.observableContext,
+    derivedPropertyExpressionState.queryBuilderState.observableContext,
   );
 };
 
 export class QueryBuilderDerivedPropertyExpressionState {
   queryBuilderState: QueryBuilderState;
   path: string;
   title: string;
-  readonly derivedProperty: DerivedProperty;
-  readonly propertyExpression!: AbstractPropertyExpression;
-  readonly parameters: VariableExpression[] = [];
+  propertyExpression: AbstractPropertyExpression;
```

Comment:
why are we removing `readonly` for these? They are fixed like this purposely so we control how they got modified.

Reply:
that wasn't on purpose. Will add them back